eventsources/sources/awssqs/start.go

```diff
@@ -112,6 +103,15 @@ func (el *EventListener) StartListening(ctx context.Context, dispatch func([]byt
 		messages, err := fetchMessages(ctx, sqsClient, *queueURL.QueueUrl, 10, sqsEventSource.WaitTimeSeconds)
 		if err != nil {
 			log.Errorw("failed to get messages from SQS", zap.Error(err))
+			awsError, ok := err.(awserr.Error)
+			if ok && awsError.Code() == "ExpiredToken" && el.SQSEventSource.SessionToken != nil {
+				log.Info("credentials expired, reading credentials again")
+				newSqsClient, err := el.createSqsClient()
+				if newSqsClient != nil && err != nil {
```

Comment:
I assume it should be `err == nil`, right? Please also print the log if err is not nil.

All the reset looks good to me.

Reply:
Oops, that was a stupid mistake. Fixed now.